warehouse/internal/repo/upload.go

```diff
@@ -1357,3 +1357,68 @@ func (u *Uploads) update(
 	}
 	return nil
 }
+
+func (u *Uploads) GetFirstAbortedUploadsInContinuousAborts(ctx context.Context, workspaceId string) ([]model.FirstAbortedUploadResponse, error) {
+	outputColumns := "id, source_id, destination_id, created_at, first_event_at, last_event_at"
+
+	stmt := fmt.Sprintf(`
+	WITH wh_uploads_with_last_successful_upload AS (
+		SELECT 
+			%s,
+			status,
+        	MAX(CASE WHEN status = 'exported_data' THEN created_at ELSE NULL END) OVER (PARTITION BY destination_id) AS last_successful_upload
+		FROM 
+			`+uploadsTableName+`
+		WHERE workspace_id = $1
+		AND created_at >= NOW() - INTERVAL '30 day'
+	)
+	SELECT %s
+	FROM (
+		SELECT 
+			*,
+			ROW_NUMBER() OVER (PARTITION BY destination_id ORDER BY created_at) as row_number
+		FROM wh_uploads_with_last_successful_upload
+		WHERE status = 'aborted'
+		AND (last_successful_upload IS NULL OR created_at > last_successful_upload)
+	) AS q
+	WHERE q.row_number = 1 
+	`, outputColumns, outputColumns)
+
+	rows, err := u.db.QueryContext(ctx, stmt, workspaceId)
+	if err != nil {
+		return nil, fmt.Errorf("first aborted upload in a series of continues aborts info: %w", err)
+	}
```

Comment:
We need to close the rows as well in defer.
```suggestion
	rows, err := u.db.QueryContext(ctx, stmt, workspaceID, queryInterval)
	if err != nil {
		return nil, fmt.Errorf("first aborted upload in a series of continues aborts info: %w", err)
	}
	defer func() { _ = rows.Close() }()
```